### ISSUE TITLE
GPG pin cache not working while running the Authenticator

### ISSUE THREAD

**Issue by xoxys** [Yubico/yubioath-desktop#617]
- Yubico Authenticator version: 5.0.4
- Operating system and version: Fedora 32
- YubiKey model and version: Yubikey 5c (5.1.2)
- Bug description summary: 

If the Yubico Authenticator app is running the gpg card will "lose" the pin on every TOTP refresh cycle and I have to enter the pin again for e.g. signing tasks. If the Authenticator app is not running the pip will be cached as long as the Yubikey was not unplugged.

Steps to reproduce
---

1. Start the Yubico Authenticator
2. gpg signing e.g. `gpg --sign test.txt` (you will be prompted for your PIN)
3. Wait for some seconds till new TOTP codes were generated
4. Run  `gpg --sign test.txt` again and you should also get the PIN prompt again

Expected result
---
Gpg PIN cache is still working while running the Authenticator app.

**Comment by xoxys** [comment]
Looks like this is related to https://github.com/Yubico/yubikey-manager/issues/35 ...

**Comment by dagheyman** [comment]
Thanks for the report. This is a limitation in GPG, that grabs the YubiKey in exclusive mode. Some further explanation is available here as well: https://github.com/Yubico/yubikey-manager/pull/331#issue-394019427

**Comment by xoxys** [comment]
Thanks for the link. If I got it right, there is currently no workaround beside patching GnuPG?

**Comment by dagheyman** [comment]
> Thanks for the link. If I got it right, there is currently no workaround beside patching GnuPG?

Correct. I've seen the patched version distributed in GPGTools for mac and in this package for Arch Linux https://aur.archlinux.org/packages/gnupg-scdaemon-shared-access/

**Comment by dagheyman** [comment]
Actually, even the patched GPG would not help with the PIN caching, since a new application (OATH) have been opened on the YubiKey and the OpenPGP PIN is then invalidated. It does allow a different application to access the device though.

**Comment by xoxys** [comment]
Phew... then I have to deal with it, somehow. Thanks again

**Comment by dainnilsson** [comment]
Yubico Authenticator 6.0 has now been released and uses a new codebase. As such, this issue has been marked with the `legacy` label, and will be automatically closed in 7 days. If this issue is still relevant to Yubico Authenticator 6, please comment on the issue saying so, and it will be kept open (or be re-opened). Sorry for the inconvenience!

**Comment by dgivens** [comment]
This is still an issue with Authenticator 6. This should at least be documented.
